### PR TITLE
perf: drop browserify

### PR DESCRIPTION
**Pull request by ThaUnknown**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[X] Other, please explain:

**What changes did you make? (Give an overview)**
Replaced browserify with webpack.
Why webpack? Because I couldn't get esbuild/rollup to do what I wanted without writing custom plugins, too much time.

Why ditch browserify?
- bad polyfills
- lack of customizability
- unnecessary polyfills [over 500kb!!!!!]

What actually changed?
- process-browserify was replaced with a modified version of [mathiasvr's process-fast](https://github.com/mathiasvr/process-fast/), modified because it didn't ponyfill queueMicrotask
- zlib-browserify was replaced with pako [for chromeapp] which is a LOT faster [again thanks for mathiasvr for figuring this out]
- added sourcemaps [wooo \o/] and ditched build-debug for that reason
- output is now an ESM module

TLDR: end result is a lot faster, a lot smaller and easier to debug

**Which issue (if any) does this pull request address?**
Oh my... it pretty much removes all those useless setTimeout calls in process.nextTick
#2351 #2191 #1568 #1026 [kind of? not really?] #903 [partial as this is also hashing, but can safely be closed]
**Is there anything you'd like reviewers to focus on?**
I dont quite understand #1965 but I feel like it could be addressed here too? explain pls?

### CODE REVIEW

File: dist/webtorrent.chromeapp.js.LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*
```
Comment (DiegoRBaquero) [inline dist/webtorrent.chromeapp.js.LICENSE.txt:1]: Can we not generate these LICENSE files?

File: dist/webtorrent.min.js.LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/*!
```
Comment (DiegoRBaquero) [inline dist/webtorrent.min.js.LICENSE.txt:1]: same

File: dist/webtorrent.chromeapp.js.LICENSE.txt (hunk 1 above)
Comment (ThaUnknown) [reply dist/webtorrent.chromeapp.js.LICENSE.txt:1]: is that.... a good idea?[ I don't really know much about copyright](https://github.com/ThaUnknown/jassub/issues/11) so I left it in

**Comment by DiegoRBaquero** [comment]
Given the new `dist` folder, a lot of external usage might break. So it's good about this being v2, any other consequences you can think of?

File: dist/webtorrent.chromeapp.js.LICENSE.txt (hunk 1 above)
Comment (DiegoRBaquero) [reply dist/webtorrent.chromeapp.js.LICENSE.txt:1]: We've never had it, never had any issues, we can include them once some copyright nerd requires them (?)

Comment (ThaUnknown) [reply dist/webtorrent.chromeapp.js.LICENSE.txt:1]: maybe I could generate only one license file instead of both?

**Comment by ThaUnknown** [comment]
> Given the new `dist` folder, a lot of external usage might break. So it's good about this being v2, any other consequences you can think of?

chromeapp.js might be broken? but I don't know how to even test that as there are no webtorrent chromeapps in the world I don't think, from what I tested all of webtorrent works fine when I ran some random local tests. Other people's browserify configs wont work anymore, but that's because it doesn't support ESM, not my fault. Since we don't do setTimeouts for process.nextTick some events might potentially run in different orders if some1 was to say run multiple webtorrent instances in the same window, but its a very minor thing and it wouldnt cause any errors, a good example is [this test](https://github.com/ThaUnknown/bittorrent-protocol/commit/e87a4ab052067e3ad9e341373faabf2aba5187a9#diff-61b342b8e5ffd8d1a2485be9eb3785549ecb329ac5d3d236c233656d8eaecef3R60)

File: dist/webtorrent.chromeapp.js.LICENSE.txt (hunk 1 above)
Comment (ThaUnknown) [reply dist/webtorrent.chromeapp.js.LICENSE.txt]: @DiegoRBaquero done

**Comment by ThaUnknown** [comment]
I'm debating if the _polyfills should be in /lib or their own folder

**Comment by ThaUnknown** [comment]
image
image
welp all tests pass so that's great, we still don't have sw tests as airtap doesn't do static files I don't think, idk what to do about that

**Comment by ThaUnknown** [comment]
@DiegoRBaquero good news, unlike the `master` branch's `webtorrent.chromeapp.js` this one actually works XD at least I think so
TLDR this is mostly useless as in a few months chrome is deprecating v2 so no socket access, and I couldn't ACTUALLY get it to work as I would need to publish some package, create a security key and some other shit, which I can't do, but IN THEORY this should work fine
I ran this code:
```js
import WebTorrent from './webtorrent.chromeapp.js'
localStorage.debug = '*'
const noop = () => {}
chrome.sockets = {
  tcpServer: {
    create: noop
  },
  udp: {
    create: noop
  },
  tcp: {
    create: noop
  }
}
chrome.dns = {
  resolve: noop
}

const client = new WebTorrent()
console.log(client)
fetch('http://webtorrent.io/torrents/sintel.torrent').then(async res => {
  const blob = await res.blob()
  const tor = client.add(blob, console.log)
  tor.on('metadata', console.warn)
  tor.on('_infoHash', console.log)
  console.log(tor)
})
```
had to make noops for the network stuff as I don't have access to it, and everything seems to work fine, metadata and ready never fire, because it needs to wait for ConnPool to connect to do so, which is noop, but metadata etc works, nothing errors on evaluation unlike the `master` branch one.
debug:
image
seems good, supposedly brave will be keeping v2 manifest, so sockets might work there

**Comment by ThaUnknown** [comment]
@DiegoRBaquero any other concerns?

**Comment by ThaUnknown** [comment]
notably `pako` can be replaced with `fflate` for even faster speeds and lower bundle sizes, but I'm not certain about compatibility with `zlib`

**Review by DiegoRBaquero — APPROVED** [review]
